### ISSUE TITLE
Use custom staging dir function for Houdini

### ISSUE THREAD

**Issue by MustafaJafar** [ynput/ayon-houdini#21]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

Use custom staging dir function for Houdini.

### How would you imagine the implementation of the feature?

Not sure, will update it later.
I think it's related to ayon core settings:
```
ayon+settings://core/tools/publish/custom_staging_dir_profiles
```

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

### Additional context:

I found this on the internal request description:
```
Based on  transient dir function pipeline.publish.lib.get_transient_dir_info should be
 used everywhere local render folders are used.
```

**Comment by BigRoy** [comment]
With the epic to go "lower-level" over Houdini and allowing more freedom/granular control by the studio and artists I feel taking control over the output paths for render ROPs might be counter-intuitive in that goal. @antirotor @dee-ynput - what do you think?

As far as I know this 'custom staging dir' intent has to reasons:
- Define a custom temp directory to render from settings (which seems the opposite of our goals for Houdini?) for the pyblish publishing logic.
- Define whether the render directory is 'persistent' or not, but as soon as we don't use custom staging dirs that kind of becomes irrelevant maybe? Although it does get in the realm of the explicit cleanup plug-in where the source render files get removed on publish?

**Comment by antirotor** [comment]
Hard to say - it depends on "inheritance". The user story:

Publishing large caches that needs to be stored before published on another (larger) local drive. You might want to set it per product type on Settings level, you might want to override it as Houdini artist locally with some tools, both might want to influence how cleanup is done after publishing. Is it really solve by "lower-level" or is it just another level?

**Comment by ralphmeijerambassadors** [comment]
Just chiming in to this discussion after I got forwarded by Roy to this discussion from the discord server. 

I think I agree with antirotor. It would be nice to be able to set the defaults per product type at the settings level, maybe even on the representation level? So alembic, bgeo, render etc. Ideally we would like to have it organised without too much user input in my opinion. Currently we have set some default output path/folders for our rops to make them a bit more organized which follows a template like the following: $hip/cache/{rop_type}(alembic, filecache,etc)/{cache_name}/{cache_version}/{cache_name}_{cache_version}.{ext}. But users are still able to step away from this when the want. However that's barely the case.

This would make things a bit more organized for artists instead of we have currently when everything is in one pyblish folder.

**Comment by BigRoy** [comment]
> `$hip/cache/{rop_type}(alembic, filecache,etc)/{cache_name}/{cache_version}/{cache_name}_{cache_version}.{ext}`

Thanks - in that context what is "cache name" and "cache version" if we were to expose that on e.g. a default Alembic ROP or alike (which does not have the notion of 'version' unlike e.g. a filecache node? Also, is "cache name" just `$OS`?

**Comment by ralphmeijerambassadors** [comment]
Yes in a way you can say cache name is $OS. However in the filecache node you have a base name and version to construct the output path.  We applied some of that same logic to the other rops . So with a integer version attribute an artist is able to set the version. Maybe this is all less important for the publishing aspect. However it would be nice to be able to submit caches to the farm without publishing for internal use by an artist, and then publish the cache manually when the artist validated the cache manually. This is however related to a whole other issue which is probably not handy to discuss here :)

**Comment by MustafaJafar** [comment]
Hello @ralphmeijerambassadors, 
Thanks for your input. 
I've a feeling that you might be mixing staging and publish directories. 
- **Staging directory**:  is the root/base output directory for ROP nodes. As far as I know, the publish system doesn't care about how organized the output paths. However, we tend to organize them for the users to have a clean work directory. staging directory in AYON are just temporary location therefore being too picky about organizing them can be a waste of time (although the system allows you to set template for staging directories which is this issue about). 
- **Publish directory**: the end location where the output files will live. you actually can be too picky about organizing publish paths where you can use product name, product version, and even representation names.

> However it would be nice to be able to submit caches to the farm without publishing for internal use by an artist, and then publish the cache manually when the artist validated the cache manually.

This is currently supported by the system but it works in a different way than you are describing.
- You don't need to submit the publish job (for an existing render) to deadline as you can just open publisher UI in Houdini and set the render target to `Use existing frames (local)`
![Image](https://github.com/user-attachments/assets/591b45a2-b8d3-4628-b528-578cd55251e4)
- You can publish existing render using tray publisher. 
- You can submit publish job for existing render using tray publisher

**Comment by MustafaJafar** [comment]
This issue requires more info: So, here it's

Custom staging directories for Houdini products are typically fetched on creating the instance unlike in other DCCs where staging directory is computed on extracting phase during publishing. 

Custom staging directories need default values that will be used if no custom staging directory profile is met. 

Based on our last internal discussion, we want to have a toggle so that we can disable managing staging directories via custom staging directories.

These points were considered and added to this PR https://github.com/ynput/ayon-houdini/pull/200
